### ISSUE TITLE
More intuitive search by name+location

### ISSUE THREAD

**Issue by raindropsfromsky** [osmandapp/Osmand#6889]
Often a search is not local: A user looks for a particular restaurant or POI in another location or locality. This is more common in metro suburbs.

For example, if I try to search for-
Barista, Indiranagar

OsmAnd fails to find Barista. Instead, it finds the **Indiranagar** locality and then gives me a long list of roads in Indiranagar.
![Barista, Indiranagar](https://user-images.githubusercontent.com/9047168/57068844-8b81c800-6cf0-11e9-8f4c-dc69a913cfb9.jpg)
In other words, OsmAnd fails to understand that I want to find the Barista shop located in Indiranagar.

But if I search for "Barista" alone, OsmAnd gives me all locations of Barista, across the globe!!
![Barista](https://user-images.githubusercontent.com/9047168/57069399-2af38a80-6cf2-11e9-97ba-6c19dd6faaab.jpg)
The long list only shows me the distance from "here", and direction.
From that, I am supposed to guess which location belongs to Indiranagar.
Thus I still don't have a clue which Barista is in Indiranagar.

Clearly, this display is not user-friendly.

**Desired:** Let the user specify a comma-separated combination such as "A, B". 

OsmAnd should understand that A refers to a POI, and B refers to a location/area/suburb/city. OsmAnd should list only those results that satisfy both POI and location criteria.

*********
BTW the fifth listing in the first screenshot points to a rather strange address: 
"1st floor, 9th A main, Indiranagar"

Now why is the first floor of **an unspecified building** on 9th A Main road an important POI??.

**Comment by mutipg** [comment]
Yes, searching in OsmAND is very unintuitive. IMO nominatim search results are better than OsmAND. I was searching for POI named "Biedronka" in city "Banino" and place I searched for is on the 8th position on the search result (picture below). Search result doesn't show city and street names for each POI so it is very hard to select correct POI to navigate.

![Zrzutekranu_2020-11-20-07-21-11-906](https://user-images.githubusercontent.com/10333015/99767554-7dc6aa00-2b03-11eb-9dda-78928b90f9d9.jpeg)

**Comment by gittyhub888** [comment]
I am all in favour of this enhancement request.
It should not be too hard to implement, as searches by street name already show the location/city to differentiate the streets found. 
All we need is the added info address and city in the POI search results.

Here is another use case: Imagine you have a date at the Mc Donald's in Paris, rue de Rennes.
A search will come up with tons of Mc Donald's restaurants and you have no way of knowing where any of these are (there are 19 at least in Paris alone), before actually selecting one and browsing to the destination on the map!!!
Definitely not user friendly. 
This is not a nice to have, this is really about the usability of this product. 
This is basic functionality that most other guidance systems have.
Please implement this.
